examples/src/components/DrillWithExternalDataExample.jsx

```diff
@@ -0,0 +1,378 @@
+                    .then(
+                        ({ results }) => {
+                            this.setState({
+                                employee3rdPartyData: {
+                                    isLoading: false,
+                                    error: null,
+                                    data: {
+                                        ...results[0],
+                                        id: employee.id,
+                                        gender
+                                    }
+                                }
+                            });
+                        },
+                        (error) => {
+                            this.setState({
+                                employee3rdPartyData: {
+                                    error,
+                                    isLoading: false,
+                                    data: null
+                                }
+                            });
+                        }
+                    );
+            });
+    }
+
+    render() {
+        const { state, location } = this.state;
+
+        const averageDailySalesMeasure = {
+            measure: {
+                localIdentifier: 'averageDailyTotalSales',
+                definition: {
+                    measureDefinition: {
+                        item: {
+                            identifier: averageDailyTotalSales
+                        }
+                    }
+                },
+                alias: 'Average Sales'
+            }
+        };
+        const stateAttribute = {
+            visualizationAttribute: {
+                localIdentifier: 'locationState',
+                displayForm: {
+                    identifier: locationStateDisplayFormIdentifier
+                }
+            }
+        };
+        const stateTable = (
+            <div style={{ height: 200 }} className="s-state-table" >
+                <Table
+                    projectId={projectId}
+                    measures={[averageDailySalesMeasure]}
+                    attributes={[stateAttribute]}
+                    drillableItems={[{
+                        identifier: locationStateDisplayFormIdentifier
+                    }]}
+                    onFiredDrillEvent={this.onStateDrill}
+                />
+            </div>
+        );
+        const employeeNameAttribute = {
+            visualizationAttribute: {
+                localIdentifier: 'employeeName',
+                displayForm: {
+                    identifier: employeeNameIdentifier
+                }
+            }
+        };
+        const locationNameAttribute = {
+            visualizationAttribute: {
+                localIdentifier: 'locationName',
+                displayForm: {
+                    identifier: locationNameDisplayFormIdentifier
+                }
+            }
+        };
+        const employeeTableFilters = [];
+        if (state) {
+            employeeTableFilters.push({
```

Comment:
I'd extract this into some kind of a function, such as addPositiveFilter(filterArray, dfIdentifier, elementUris), it's used multiple times through the code